test/common/run-tests

```diff
@@ -107,6 +108,21 @@ class Test:
             self._print_test(print_tap, skip_reason=skip_reason)
             return None, 0
 
+        # If the test was marked with @todo then...
+        if self.todo is not None:
+            todo_reason, flaky = self.todo
+            if self.returncode == 0 and not flaky:
+                # The test passed, but it shouldn't have.
+                self.returncode = 1  # that's a fail
+                self._print_test(print_tap, todo_reason=f'# expected failure: {todo_reason}')
+                return None, 1
+            else:
+                # The test either failed (always OK) or passed (and flaky=True)
+                # Outputs  'not ok 1 test # TODO ...'
+                #      or  'ok 1 test # TODO ...'
+                self._print_test(print_tap, todo_reason=f'# TODO {todo_reason}')
```

Comment:
This keeps printing the `# TODO` even for successful (flaky) tests. The TAP spec doesn't mention this case, and my recent [log.html change](https://github.com/cockpit-project/bots/pull/4180) doesn't recognize that either, but AFAICS it *should* be fine. It will just appear in the HTML as a normal successful test.